### PR TITLE
Improve docs of constant liar

### PR DESCRIPTION
**Pull request by hvy**
## Motivation

See below description.

## Description of the changes

Adds a note to the constant liar docs that zombie trials also will be considered.

### CODE REVIEW

**Review by keisuke-umezawa — APPROVED** [review]
LGTM!

File: optuna/samplers/_tpe/sampler.py [hunk 1]
```diff
@@ -199,6 +199,12 @@ def objective(trial):
             If :obj:`True`, penalize running trials to avoid suggesting parameter configurations
             nearby.
 
+            .. note::
+                Abnormally terminated trials are often in the running state.
```
Comment (Crissman) [inline optuna/samplers/_tpe/sampler.py:203]:
```suggestion
                Abnormally terminated trials often leave behind a record with a state of `RUNNING` in the storage.
```

File: optuna/samplers/_tpe/sampler.py [hunk 2]
```diff
@@ -199,6 +199,12 @@ def objective(trial):
...
+                Such zombie trials will be penalized and taken into account during optimization.
```
Comment (Crissman) [inline optuna/samplers/_tpe/sampler.py:204]:
```suggestion
                Such "zombie" trial parameters will be avoided by the constant liar algorithm during subsequent sampling.
```

File: optuna/samplers/_tpe/sampler.py [hunk 3]
```diff
@@ -199,6 +199,12 @@ def objective(trial):
...
+                When using an :class:`~optuna.storages.RDBStorage`, it is possible to enable the
+                ``heartbeat_interval`` to automatically fail such zombie trials.
```
Comment (Crissman) [inline optuna/samplers/_tpe/sampler.py:206]:
```suggestion
                ``heartbeat_interval`` to change the records for abnormally terminated trials to `FAIL`.
```

**Review by Crissman — CHANGES_REQUESTED** [review]
Thanks for the PR! Left some edits for clarity.

**Comment by hvy** [comment]
@Crissman thanks for the review, I applied your suggestions. PTAL. 🙇

**Review by Crissman — APPROVED** [review]
LGTM.

**Pull request merged**